prov/tcp/src/xnet_progress.c

```diff
@@ -241,27 +241,29 @@ void xnet_recv_saved(struct xnet_xfer_entry *saved_entry,
 	xnet_free_xfer(progress, rx_entry);
 }
 
-void xnet_update_pollflag(struct xnet_ep *ep, short pollflag, bool set)
+static int xnet_update_pollflag(struct xnet_ep *ep, short pollflag, bool set)
 {
 	struct xnet_progress *progress;
+	int ret;
 
 	progress = xnet_ep2_progress(ep);
 	assert(xnet_progress_locked(progress));
 	if (set) {
 		if (ep->pollflags & pollflag)
-			return;
+			return 0;
 
 		ep->pollflags |= pollflag;
 	} else {
 		if (!(ep->pollflags & pollflag))
-			return;
+			return 0;
 
 		ep->pollflags &= ~pollflag;
 	}
 
-	ofi_dynpoll_mod(&progress->epoll_fd, ep->bsock.sock,
-			ep->pollflags, &ep->util_ep.ep_fid.fid);
+	ret = ofi_dynpoll_mod(&progress->epoll_fd, ep->bsock.sock,
+			      ep->pollflags, &ep->util_ep.ep_fid.fid);
 	xnet_signal_progress(progress);
+	return ret;
```

Comment:
In the case of epoll, if we need to set a flag and fail, we have the potential to hang the app by missing events.  If we're clearing the flag and can't, I think that's recoverable.  We may generate events that we don't want, which can be inefficient, but it's not fatal.  Plus, we'll retry clearing the flag later (until we eventually succeed).

Whether we want to ignore errors clearing the flag is debatable.  The larger question is the behavior when we can't ignore the errors, and disabling the ep seems reasonable.